plugins/base/src/main/kotlin/transformers/documentables/KotlinArrayDocumentableReplacerTransformer.kt

```diff
@@ -0,0 +1,63 @@
+
+class KotlinArrayDocumentableReplacerTransformer(context: DokkaContext):
+    DocumentableReplacerTransformer(context) {
+
+    private fun Documentable.isJVM() =
+        sourceSets.any{ it.analysisPlatform == Platform.jvm }
+
+    override fun processGenericTypeConstructor(genericTypeConstructor: GenericTypeConstructor): AnyWithChanges<GenericTypeConstructor> =
+        genericTypeConstructor.takeIf { genericTypeConstructor.dri == DRI("kotlin", "Array") }
+            ?.let {
+                with(it.projections.firstOrNull() as? Variance<Bound>) {
+                    with(this?.inner as? GenericTypeConstructor) {
+                        when (this?.dri) {
+                            DRI("kotlin", "Int") ->
+                                AnyWithChanges(
+                                    GenericTypeConstructor(DRI("kotlin", "IntArray"), emptyList()),
+                                    true)
+                            DRI("kotlin", "Boolean") ->
+                                AnyWithChanges(
+                                    GenericTypeConstructor(DRI("kotlin", "BooleanArray"), emptyList()),
+                                    true)
+                            DRI("kotlin", "Float") ->
+                                AnyWithChanges(
+                                    GenericTypeConstructor(DRI("kotlin", "FloatArray"), emptyList()),
+                                    true)
+                            DRI("kotlin", "Double") ->
+                                AnyWithChanges(
+                                    GenericTypeConstructor(DRI("kotlin", "DoubleArray"), emptyList()),
+                                    true)
+                            DRI("kotlin", "Long") ->
+                                AnyWithChanges(
+                                    GenericTypeConstructor(DRI("kotlin", "LongArray"), emptyList()),
+                                    true)
+                            DRI("kotlin", "Short") ->
+                                AnyWithChanges(
+                                    GenericTypeConstructor(DRI("kotlin", "ShortArray"), emptyList()),
+                                    true)
+                            DRI("kotlin", "Char") ->
+                                AnyWithChanges(
+                                    GenericTypeConstructor(DRI("kotlin", "CharArray"), emptyList()),
+                                    true)
+                            DRI("kotlin", "Byte") ->
+                                AnyWithChanges(
+                                    GenericTypeConstructor(DRI("kotlin", "ByteArray"), emptyList()),
+                                    true)
+                            else -> null
+                        }
+                    }
+                }
+            }
+            ?: super.processGenericTypeConstructor(genericTypeConstructor)
+
+    override fun processModule(module: DModule): AnyWithChanges<DModule>  =
+        if(module.isJVM())
```

Comment:
I think that we can narrow down the scope of searching only to java classes, but watch for mixed packages (where java and kotlin classes are in 1 package).

The easiest way to check language would be to look at the source like i did here: https://github.com/Kotlin/dokka/pull/2048/files